In [1]:
!pip install numpy
!pip install pandas



import numpy as np
import pandas as pd



Defaulting to user installation because normal site-packages is not writeable
Defaulting to user installation because normal site-packages is not writeable


<h3>Converting Database into DataFrame</h3>
<hr>

In [2]:
drive = pd.read_csv('uk_road_accident.csv')

In [3]:
drive

Index Accident_Severity Accident Date   Latitude  \
0       200701BS64157           Serious      5/6/2019  51.506187   
1       200701BS65737           Serious      2/7/2019  51.495029   
2       200701BS66127           Serious    26-08-2019  51.517715   
3       200701BS66128           Serious    16-08-2019  51.495478   
4       200701BS66837            Slight      3/9/2019  51.488576   
...               ...               ...           ...        ...   
660674  201091NM01760            Slight    18-02-2022  57.374005   
660675  201091NM01881            Slight    21-02-2022  57.232273   
660676  201091NM01935            Slight    23-02-2022  57.585044   
660677  201091NM01964           Serious    23-02-2022  57.214898   
660678  201091NM02142           Serious    28-02-2022  57.575210   

                   Light_Conditions           District Area  Longitude  \
0             Darkness - lights lit  Kensington and Chelsea  -0.209082   
1                          Daylight  Kensington and Chelsea  -0.173647   
2       Darkness - lighting unknown  Kensington and Chelsea  -0.210215   
3                          Daylight  Kensington and Chelsea  -0.202731   
4             Darkness - lights lit  Kensington and Chelsea  -0.192487   
...                             ...                     ...        ...   
660674                     Daylight                Highland  -3.467828   
660675       Darkness - no lighting                Highland  -3.809281   
660676                     Daylight                Highland  -3.862727   
660677       Darkness - no lighting                Highland  -3.823997   
660678                     Daylight                Highland  -3.895673   

        Number_of_Casualties  Number_of_Vehicles Road_Surface_Conditions  \
0                          1                   2                     Dry   
1                          1                   2             Wet or damp   
2                          1                   3                     Dry   
3                          1                   4                     Dry   
4                          1                   2                     Dry   
...                      ...                 ...                     ...   
660674                     2                   1                     Dry   
660675                     1                   1            Frost or ice   
660676                     1                   3            Frost or ice   
660677                     1                   2             Wet or damp   
660678                     1                   1             Wet or damp   

                 Road_Type Urban_or_Rural_Area     Weather_Conditions  \
0       Single carriageway               Urban     Fine no high winds   
1       Single carriageway               Urban  Raining no high winds   
2                      NaN               Urban                    NaN   
3       Single carriageway               Urban     Fine no high winds   
4                      NaN               Urban                    NaN   
...                    ...                 ...                    ...   
660674  Single carriageway               Rural     Fine no high winds   
660675  Single carriageway               Rural     Fine no high winds   
660676  Single carriageway               Rural     Fine no high winds   
660677  Single carriageway               Rural     Fine no high winds   
660678    Dual carriageway               Rural  Snowing no high winds   

                                Vehicle_Type  
0                                        Car  
1                                        Car  
2                      Taxi/Private hire car  
3       Bus or coach (17 or more pass seats)  
4                              Other vehicle  
...                                      ...  
660674                                   Car  
660675                                   Car  
660676                                   Car  
660677                 Motorcycle over 500cc  
660

<h3>Questions & Insights</h3>
<hr>

<h3>Question NO.1</h3>
<h4> In the accidents that happend which of the Accident_Severity has the most counts?</h4>

In [4]:
drive['Accident_Severity'].value_counts()

Accident_Severity
Slight     563801
Serious     88217
Fatal        8661
Name: count, dtype: int64

<h3>Insights NO.1</h3>
<h4>Accorging to the data Slight damage has the most number</h4>

<h3>Question NO.2</h3>
<h4>What weather had the most serious vihicle accident that happend?</h4>

In [5]:
weather = drive.groupby(['Weather_Conditions', 'Accident_Severity']).size()

In [6]:
weather

Weather_Conditions     Accident_Severity
Fine + high winds      Fatal                   175
                       Serious                1245
                       Slight                 7134
Fine no high winds     Fatal                  7100
                       Serious               72046
                       Slight               441739
Fog or mist            Fatal                    82
                       Serious                 483
                       Slight                 2963
Other                  Fatal                   165
                       Serious                1801
                       Slight                15184
Raining + high winds   Fatal                   145
                       Serious                1261
                       Slight                 8209
Raining no high winds  Fatal                   848
                       Serious                9468
                       Slight                69380
Snowing + high winds   Fatal             

<h3>Insights NO.2</h3>
<h4>The Fine no high winds weather has the most seriouse accident with 72046 total incident</h4>

<h3>Question NO.3</h3>
<h4>In the Road Surface Conditions what is prone to accident?</h4>

In [7]:
drive['Road_Surface_Conditions'].value_counts()

Road_Surface_Conditions
Dry                     447821
Wet or damp             186708
Frost or ice             18517
Snow                      5890
Flood over 3cm. deep      1017
Name: count, dtype: int64

<h3>Insights NO.3</h3>
<h4>The Dry surface has the highest incident</h4>

<h3>Question NO.4</h3>
<h4>What kind of vehicle has the highest accident record</h4>

In [8]:
drive['Vehicle_Type'].value_counts()

Vehicle_Type
Car                                      497992
Van / Goods 3.5 tonnes mgw or under       34160
Bus or coach (17 or more pass seats)      25878
Motorcycle over 500cc                     25657
Goods 7.5 tonnes mgw and over             17307
Motorcycle 125cc and under                15269
Taxi/Private hire car                     13294
Motorcycle over 125cc and up to 500cc      7656
Motorcycle 50cc and under                  7603
Goods over 3.5t. and under 7.5t            6096
Other vehicle                              5637
Minibus (8 - 16 passenger seats)           1976
Agricultural vehicle                       1947
Pedal cycle                                 197
Data missing or out of range                  6
Ridden horse                                  4
Name: count, dtype: int64

<h3>Insights NO.4</h3>
<h4>The Car has the most accident record</h4>

<h3>Question NO.5</h3>
<h3>Which of the two Urban and Rural have more accident records?</h3>

In [9]:
drive['Urban_or_Rural_Area'].value_counts()

Urban_or_Rural_Area
Urban          421663
Rural          238990
Unallocated        11
Name: count, dtype: int64

<h3>Insights NO.5</h3>
<h4>The Urban has more accident record than Rural</h4>

<h3>Question NO.6</h3>
<h4>Which part in the UK have the highest number of recorded road accidents?</h4>

In [10]:
drive['District Area'].value_counts()

District Area
Birmingham            13491
Leeds                  8898
Manchester             6720
Bradford               6212
Sheffield              5710
                      ...  
Berwick-upon-Tweed      153
Teesdale                142
Shetland Islands        133
Orkney Islands          117
Clackmannanshire         91
Name: count, Length: 422, dtype: int64

<h3>Insight no.6</h3>
<h4>According to the computation Birmingham has the most accidents that happend </h4>

<h3>Question NO.7</h3>
<h4>How many person is usually involve in a road accident?</h4>

In [11]:
drive['Number_of_Casualties'].mode()

0    1
Name: Number_of_Casualties, dtype: int64

<h3>Insights.No.7</h3>
<h4>Mostly only one person is involved on a road accident</h4>

<h3>Question NO.8</h3>
<h4>What weather has the least accident record?</h4>

In [12]:
drive['Weather_Conditions'].value_counts()

Weather_Conditions
Fine no high winds       520885
Raining no high winds     79696
Other                     17150
Raining + high winds       9615
Fine + high winds          8554
Snowing no high winds      6238
Fog or mist                3528
Snowing + high winds        885
Name: count, dtype: int64

<h3>Insights NO.8</h3>
<h4>The Snowing + high winds has the least accident record</h4>

<h3>Question NO.9</h3>
<h4>How many vehicle mostly have an accident in one scene?</h4>

In [13]:
drive['Number_of_Vehicles'].value_counts()

Number_of_Vehicles
2     391995
1     200787
3      52806
4      11300
5       2464
6        758
7        296
8        147
9         57
10        27
11        12
14         9
12         7
13         6
16         4
28         1
15         1
32         1
19         1
Name: count, dtype: int64

<h3>Insights NO.9</h3>
<h4>2 Vehicles has the most record of an accident in one scene</h4>

<h3>Question NO.10</h3>
<h4>In this two different kind of vehicle <b>(Agricultural vehicle)</b>, <b>(Van)</b> which of the two has more accident in a single carriageway?</h4>

In [14]:
rr= drive.groupby(['Vehicle_Type','Road_Type']).size()

In [15]:
rr

Vehicle_Type                         Road_Type         
Agricultural vehicle                 Dual carriageway        293
                                     One way street           49
                                     Roundabout              129
                                     Single carriageway     1436
                                     Slip road                27
                                                           ...  
Van / Goods 3.5 tonnes mgw or under  Dual carriageway       5107
                                     One way street          723
                                     Roundabout             2294
                                     Single carriageway    25460
                                     Slip road               361
Length: 73, dtype: int64

<h3>Insights NO.10</h3>
<h4>The Van has more accident in a single carriageway than the Agriculture vehicle</h4>

<h3>Question NO.11</h3>
<h4>Which of the district areas has the least Road Surface Condition accidents?</h4>

In [16]:
DR= drive.groupby(['District Area', 'Road_Surface_Conditions']).size()

In [17]:
DR

District Area  Road_Surface_Conditions
Aberdeen City  Dry                         815
               Frost or ice                 25
               Snow                         15
               Wet or damp                 468
Aberdeenshire  Dry                         941
                                          ... 
Wyre Forest    Wet or damp                 238
York           Dry                        1324
               Frost or ice                 42
               Snow                         22
               Wet or damp                 508
Length: 2005, dtype: int64

<h3>Insights NO.11</h3>
<h4>The Aberdeen City has the least and it's a snow condition with 15 incident</h4>

<h3>Question NO.12</h3>
<h4>Which Area has the least accident?</h4>

In [18]:
drive['District Area'].value_counts()

District Area
Birmingham            13491
Leeds                  8898
Manchester             6720
Bradford               6212
Sheffield              5710
                      ...  
Berwick-upon-Tweed      153
Teesdale                142
Shetland Islands        133
Orkney Islands          117
Clackmannanshire         91
Name: count, Length: 422, dtype: int64

<h3>Insights NO.12</h3>
<h4>The Clackmannanshire has the least accident</h4>

<h3>Question NO.13</h3>
<h4>How many accidents does the Pedal cycle have been involved during the daylight? </h4>

In [19]:
car = drive.groupby(['Light_Conditions', 'Vehicle_Type']).size()

In [20]:
car

Light_Conditions             Vehicle_Type                        
Darkness - lighting unknown  Agricultural vehicle                       19
                             Bus or coach (17 or more pass seats)      267
                             Car                                      4914
                             Goods 7.5 tonnes mgw and over             185
                             Goods over 3.5t. and under 7.5t            59
                                                                     ...  
Daylight                     Other vehicle                            4132
                             Pedal cycle                               146
                             Ridden horse                                3
                             Taxi/Private hire car                    9752
                             Van / Goods 3.5 tonnes mgw or under     24944
Length: 74, dtype: int64

<h3>Insights NO.13</h3>
<h4>There are 146 records that the Pedal cycle have been involved to a accident during the daylight</h4>

<h3>Question NO.14</h3>
<h4>What kind of road type has the least accident?</h4>

In [21]:
car = drive.groupby(['Light_Conditions', 'Vehicle_Type']).size()

In [22]:
drive['Road_Type'].value_counts()

Road_Type
Single carriageway    492143
Dual carriageway       99424
Roundabout             43992
One way street         13559
Slip road               7041
Name: count, dtype: int64

<h3>Insights NO.14</h3>
<h4>The slip road has the least accident</h4>

<h3>Question NO.15</h3>
<h4>How many car have an accident during the darkness?</h4>

In [23]:
car

Light_Conditions             Vehicle_Type                        
Darkness - lighting unknown  Agricultural vehicle                       19
                             Bus or coach (17 or more pass seats)      267
                             Car                                      4914
                             Goods 7.5 tonnes mgw and over             185
                             Goods over 3.5t. and under 7.5t            59
                                                                     ...  
Daylight                     Other vehicle                            4132
                             Pedal cycle                               146
                             Ridden horse                                3
                             Taxi/Private hire car                    9752
                             Van / Goods 3.5 tonnes mgw or under     24944
Length: 74, dtype: int64

<h3>Insights NO.15</h3>
<h4>There are 4914 cars that have been into accident during the darkness</h4>

<h3>Question NO.16</h3>
<h4>What kind of light condition has the least accident record?</h4>

In [24]:
drive['Light_Conditions'].value_counts()

Light_Conditions
Daylight                       484880
Darkness - lights lit          129335
Darkness - no lighting          37437
Darkness - lighting unknown      6484
Darkness - lights unlit          2543
Name: count, dtype: int64

<h3>Insights NO.16</h3>
<h4>Darkness - lights unlit has the least Light condition record</h4>

<h3>Question NO.17</h3>
<h4>What kind of road type has the least most record during darkness lights unlit?</h4>

In [25]:
light = drive.groupby(['Light_Conditions', 'Road_Type']).size()

In [26]:
light

Light_Conditions             Road_Type         
Darkness - lighting unknown  Dual carriageway         905
                             One way street           124
                             Roundabout               470
                             Single carriageway      4664
                             Slip road                 86
Darkness - lights lit        Dual carriageway       20242
                             One way street          3310
                             Roundabout             10528
                             Single carriageway     93337
                             Slip road               1315
Darkness - lights unlit      Dual carriageway         465
                             One way street            58
                             Roundabout               189
                             Single carriageway      1770
                             Slip road                 34
Darkness - no lighting       Dual carriageway        7460
                        

<h3>Insights NO.17</h3>
<h4>The Single carriageway has the most record of accident during the Darkness lights unlit </h4>

<h3>Question NO.18</h3>
<h3>What latitude has the most record?</h3>

In [27]:
drive['Latitude'].mode()

0    52.458798
Name: Latitude, dtype: float64

<h3>Insights NO.18</h3>
<h4>Latitude 52.458798 has the most record</h4>

<h3>Question NO.19</h3>
<h4>In a serious road accident what light condition has the most accident?</h4>

In [28]:
acl = drive.groupby(['Accident_Severity', 'Light_Conditions']).size()

In [29]:
acl

Accident_Severity  Light_Conditions           
Fatal              Darkness - lighting unknown        68
                   Darkness - lights lit            1860
                   Darkness - lights unlit            45
                   Darkness - no lighting           1612
                   Daylight                         5076
Serious            Darkness - lighting unknown       794
                   Darkness - lights lit           19130
                   Darkness - lights unlit           360
                   Darkness - no lighting           7174
                   Daylight                        60759
Slight             Darkness - lighting unknown      5622
                   Darkness - lights lit          108345
                   Darkness - lights unlit          2138
                   Darkness - no lighting          28651
                   Daylight                       419045
dtype: int64

<h3>Insights NO.19</h3>
<h4>According to the data Daylight  has the most recored of having an accident</h4>

<h3>Question NO.20</h3>
<h4>What date has the most accident that happend?</h4>

In [30]:
drive['Accident Date'].mode()

0    30-11-2019
Name: Accident Date, dtype: object

<h3>Insights NO.20</h3>
<h4>According to data records 30-11-2019 has the most accident that happend</h4>